src/views/Timeline/TimelineView.svelte

```diff
@@ -0,0 +1,244 @@
+
+  import {
+    addInterval,
+    generateDates,
+    generateTitle,
+    isCalendarInterval,
+    subtractInterval,
+  } from "../Calendar/calendar";
+
+  import { computeDateInterval, computeHours, generateHours } from "./timeline";
+  import type { TimelineConfig } from "./types";
+  import TimelineBackground from "./components/TimelineBackground.svelte";
+
+  import HorizontalGroup from "src/components/HorizontalGroup/HorizontalGroup.svelte";
+  import { Field } from "src/components/Field";
+  import { fieldToSelectableValue } from "../helpers";
+  import TimelineEntry from "./components/TimelineEntry.svelte";
+
+  import type { Optional, OptionalDataValue } from "src/lib/data";
+  import { EditNoteModal } from "src/modals/edit-note-modal";
+  import { get } from "svelte/store";
+
+  export let project: ProjectDefinition;
+  export let frame: DataFrame;
+  export let readonly: boolean;
+  export let api: ViewApi;
+
+  export let config: TimelineConfig;
+
+  export let onConfigChange: (cfg: TimelineConfig) => void;
+
+  $: ({ fields, records } = frame);
+
+  $: dateFields = fields.filter((field) => field.type === DataFieldType.Date);
+  $: startDateField =
+    dateFields.find((field) => config?.startDateField === field.name) ??
+    dateFields[0];
+  $: endDateField =
+    dateFields.find((field) => config?.endDateField === field.name) ??
+    dateFields[1] ??
+    dateFields[0];
+
+  $: booleanFields = fields.filter(
+    (field) => field.type === DataFieldType.Boolean
+  );
+  $: booleanField = fields.find((field) => config?.checkField === field.name);
+
+  let anchorDate: dayjs.Dayjs = dayjs();
+  $: interval = config?.interval ?? "week";
+  $: dateInterval =
+    interval != "day"
+      ? computeDateInterval(anchorDate, interval)
+      : computeHours(anchorDate);
+  $: title = dateInterval ? generateTitle(dateInterval) : "";
+  $: dates = dateInterval
+    ? interval == "day"
+      ? generateHours(dateInterval)
+      : generateDates(dateInterval)
+    : [];
+
+  $: _records = records.filter((a) => {
+    let startday = dayjs(a.values[startDateField!.name] as Date);
+    let endday = dayjs(a.values[endDateField!.name] as Date);
+    return (
+      startday.isAfter(dateInterval[0]) ||
+      startday.isBefore(dateInterval[1]) ||
+      endday.isAfter(dateInterval[0]) ||
+      endday.isBefore(dateInterval[1])
+    );
+  });
+
+  function handleIntervalChange(interval: string) {
+    if (isCalendarInterval(interval)) {
+      onConfigChange({ ...config, interval });
+    }
+  }
+  function handleCheckFieldChange(checkField: string) {
+    onConfigChange({ ...config, checkField });
+  }
+  function handleDateFieldChange(
+    dateField: string,
+    startOrEnd: "start" | "end"
+  ) {
+    let name = startOrEnd == "start" ? "startDateField" : "endDateField";
+    onConfigChange({ ...config, [name]: dateField });
+  }
+  function asOptionalBoolean(value: OptionalDataValue): boolean | Optional {
```

Comment:
```suggestion
  function asOptionalBoolean(value: Optional<DataValue>): boolean | Optional {
```

Changed this to be more generic.